src/Api/Controllers/CollectionsController.cs

```diff
@@ -170,9 +190,86 @@ private async Task<Collection> GetCollectionAsync(Guid id, Guid orgId)
-        private async Task<bool> ManageAnyCollections(Guid orgId)
+
+        public async Task<bool> CanCreateCollection(Guid orgId, Guid collectionId)
+        {
+            if (collectionId != default)
+            {
+                return false;
+            }
+
+            return await _currentContext.CreateNewCollections(orgId);
+        }
+
+        private async Task<bool> CanEditCollectionAsync(string orgId, string collectionId) =>
+            await CanEditCollectionAsync(new Guid(orgId), new Guid(collectionId));
+        private async Task<bool> CanEditCollectionAsync(Guid orgId, Guid collectionId)
+        {
+            if (collectionId == default)
+            {
+                return false;
+            }
+
+            if (await _currentContext.EditAnyCollection(orgId))
+            {
+                return true;
+            }
+
+            if (await _currentContext.EditAssignedCollections(orgId))
+            {
+                return null != _collectionRepository.GetByIdAsync(collectionId, _currentContext.UserId.Value);
+            }
+
+            return false;
+        }
+
+        private async Task<bool> CanDeleteCollectionAsync(string orgId, string collectionId) =>
+            await CanDeleteCollectionAsync(new Guid(orgId), new Guid(collectionId));
+        private async Task<bool> CanDeleteCollectionAsync(Guid orgId, Guid collectionId)
+        {
+            if (collectionId == default)
+            {
+                return false;
+            }
+
+            if (await _currentContext.DeleteAnyCollection(orgId))
+            {
+                return true;
+            }
+
+            if (await _currentContext.DeleteAssignedCollections(orgId))
+            {
+                return null != _collectionRepository.GetByIdAsync(collectionId, _currentContext.UserId.Value);
+            }
+
+            return false;
+        }
+
+        private async Task<bool> CanViewCollectionAsync(string orgId, string collectionId) =>
+            await CanViewCollectionAsync(new Guid(orgId), new Guid(collectionId));
+        private async Task<bool> CanViewCollectionAsync(Guid orgId, Guid collectionId)
+        {
+            if (collectionId == default)
+            {
+                return false;
+            }
+
+            if (await _currentContext.ViewAllCollections(orgId))
+            {
+                return true;
+            }
+
+            if (await _currentContext.ViewAssignedCollections(orgId))
+            {
+                return null != _collectionRepository.GetByIdAsync(collectionId, _currentContext.UserId.Value);
+            }
+
+            return false;
+        }
+
+        private async Task<bool> ViewAtLeastOneCollectionAsync(Guid orgId)
         {
-            return await _currentContext.ManageAssignedCollections(orgId) || await _currentContext.ManageAllCollections(orgId);
+            return await _currentContext.ViewAllCollections(orgId) || await _currentContext.ViewAssignedCollections(orgId);
```

Comment:
I would kind of have assumed `ViewAllCollection == true => ViewAssignedCollections == true`. Since AssignedCollections is a subset of AllCollections. Any specific reason this isn't true?

Reply:
It does. I don't think I understand what you're getting at. The or exists for when you can't view all, but can view assigned.